File: app/src/main/java/de/dotwee/rgb/canteen/model/proxy/MensaCallback.java
package de.dotwee.rgb.canteen.model.proxy;

import android.support.annotation.NonNull;

import java.io.IOException;
import java.util.ArrayList;

import de.dotwee.rgb.canteen.model.Item;
import okhttp3.Call;

/**
 * Created by lukas on 18.04.18.
 */

public interface MensaCallback {

    void onFailure(@NonNull Call call, @NonNull IOException e);

    void onResponse(@NonNull Call call, @NonNull ArrayList<Item> items) throws IOException;
}

File: app/src/main/java/de/dotwee/rgb/canteen/model/proxy/MensaCallback.kt
package de.dotwee.rgb.canteen.model.proxy

import de.dotwee.rgb.canteen.model.Item
import okhttp3.Call
import java.io.IOException
import java.util.*

/**
 * Created by lukas on 18.04.18.
 */

interface MensaCallback {

    fun onFailure(call: Call, e: IOException)

    @Throws(IOException::class)
    fun onResponse(call: Call, items: ArrayList<Item>)
}
